### ISSUE TITLE
False positive for Lint/ShadowedArgument when reassigning to splat variable.

### ISSUE THREAD

**Issue by reitermarkus** [rubocop-hq/rubocop#6029]
I have the following method:

```ruby
def method(*items)
  *items, last = items
  return last if items.empty?
  items
end
```

The `Lint/ShadowedArgument` cop warns “Argument items was shadowed by a local variable before it was used.”, yet when I add `items = items` before `*items, last = items`, the warning is gone.

--------

## Expected behavior

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behavior

```
Inspecting 1 file
W

Offenses:

test.rb:2:4: W: Lint/ShadowedArgument: Argument items was shadowed by a local variable before it was used.
  *items, last = items
   ^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Run RuboCop on a file containing the method mentioned above.

## RuboCop version

```
0.57.2 (using Parser 2.5.1.0, running on ruby 2.3.3 universal.x86_64-darwin17)
```
